### PR TITLE
base-images: Use new debian-iptables:v12.1.0

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**
/kind cleanup
/area dependency security release-eng

**What this PR does / why we need it**:
- Update dependents to use debian-iptables:v12.1.0
- Update guidance for updating base image tags and digests

/assign @dims @BenTheElder @cblecker 
cc: @kubernetes/release-engineering
/priority important-soon

**Special notes for your reviewer**:

Image needs to be promoted here: https://github.com/kubernetes/k8s.io/pull/839
Continuation of https://github.com/kubernetes/kubernetes/pull/90697.
Tracking issue: https://github.com/kubernetes/kubernetes/issues/90698

**Does this PR introduce a user-facing change?**:
```release-note
- base-images: Use debian-iptables:v12.1.0
```

### CODE REVIEW

**Comment by justaugustus** [comment]
/retest
(now that https://github.com/kubernetes/k8s.io/pull/839 has merged)

File: build/workspace.bzl [hunk 1]
```diff
@@ -71,7 +71,11 @@ def cri_tarballs():
             urls = mirror("https://github.com/kubernetes-incubator/cri-tools/releases/download/v%s/crictl-v%s-%s.tar.gz" % (CRI_TOOLS_VERSION, CRI_TOOLS_VERSION, arch)),
         )
 
-# Use go get -u github.com/estesp/manifest-tool to find these values
+# Use skopeo to find these values: https://github.com/containers/skopeo/releases
```
Comment (saschagrunert) [inline build/workspace.bzl:74]: Is the link to the /releases intended? If not we could point to the README (repo only url).

File: build/workspace.bzl [hunk 2]
```diff
@@ -104,16 +113,18 @@ def debian_image_dependencies():
...
+            # Ensure the digests above are updated to match a new tag
             tag = "v2.1.0",  # ignored, but kept here for documentation
         )
 
         container_pull(
             name = "debian-iptables-" + arch,
             architecture = arch,
             digest = _digest(_DEBIAN_IPTABLES_DIGEST, arch),
-            registry = "k8s.gcr.io",
+            registry = "us.gcr.io/k8s-artifacts-prod/build-image",
```
Comment (saschagrunert) [inline build/workspace.bzl:124]: Is the bucket only available there or would `gcr.io/…` work too?

**Review by saschagrunert — APPROVED** [review]
Too minor nits, otherwise 
/lgtm

File: build/workspace.bzl (hunk 1 above)
Comment (justaugustus) [reply build/workspace.bzl]: Fixed!

File: build/workspace.bzl (hunk 2 above)
Comment (justaugustus) [reply build/workspace.bzl:124]: @saschagrunert -- `gcr.io/...` would _NOT_ work.
`us.gcr.io` is one of the geo-specific endpoints images are promoted into (the others being `{eu,asia}.gcr.io).

Once the vanity domain flip (VDF) is complete (sometime in May), `k8s.gcr.io` will geo-target the appropriate endpoints.

We will then sweep references to `us.gcr.io/k8s-artifacts-prod/<staging-project>` and revert them to `k8s.gcr.io`.

More details: https://github.com/kubernetes/k8s.io/blob/master/k8s.gcr.io/Vanity-Domain-Flip.md
Tracking issue: https://github.com/kubernetes/release/issues/270

Comment (saschagrunert) [reply build/workspace.bzl:124]: That helps, thank you for the clarification 🙏

**Review by fejta — APPROVED** [review]
/approve

**Pull request merged**